src/components/interface/cbufp/cbuf_meta.h

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2012 by The George Washington University.  All rights reserved.
+ *
+ * Redistribution of this file is permitted under the GNU General
+ * Public License v2.
+ *
+ * Author: Gabriel Parmer, gparmer@gwu.edu, 2012
+ */
+
+#ifndef CBUF_META_H
+#define CBUF_META_H
+#define CBUFM_REFCNT_SZ     9
+#define CBUFM_FIRST_FLAGS_SZ      3
+#define CBUFM_SECOND_FLAGS_SZ      4
+#define CBUFM_NEXT_MASK 0xf
+#define CBUFM_PTR_MASK 0xfff
+#define CBUFM_INC_UNIT     (1<<CBUFM_FIRST_FLAGS_SZ)
+#define CBUFM_DEC_UNIT     (~(CBUFM_INC_UNIT)+1)
+#define CBUF_REFCNT_MAX   ((1<<CBUFM_REFCNT_SZ)-1)
+#define CBUF_SENDRECV_MAX  ((1<<8)-1)
+
+/*get flags*/
+#define CBUF_OWNER(meta)  ((meta)->nfo & CBUFM_OWNER)
```

Comment:
try lining up as many of these as is reasonable within the different groups.  It is hard to quickly see where the macro name ends, and the contents begin without this.